### PR TITLE
Tests for string append

### PR DESCRIPTION
**Pull request by bebehei**
This is somewhat unrelated to previous PRs, but during test runs it bugged me, that this single skipped test was there.

Additionally, I made the `string_append` function also respect empty strings and concatenate the separator only, if both values are set.

### CODE REVIEW

**Comment by tsipinakis** [comment]
Hmm looking at the behaviour of `string_append` is there a reason we're not using `g_strjoin`?

**Comment by bebehei** [comment]
g_strjoin does not work if a is NULL, as the list is ended and b is avoided

File: src/utils.c [hunk 1]
```diff
@@ -76,9 +76,9 @@ char *string_replace_all(const char *needle, const char *replacement,
 
 char *string_append(char *a, const char *b, const char *sep)
 {
-        if (!a)
+        if (!a || *a == '\0')
```
Comment (tsipinakis) [inline src/utils.c]: `a` isn't freed in this case if it's `\0` which could cause an (albeit minor) memory leak.

I have no clue *why* this function needs to free only `a` but that's how it works it's best to preserve that until we can figure out why.

Comment (bebehei) [reply src/utils.c]: > a isn't freed in this case if it's \0 which could cause an (albeit minor) memory leak.

a very good catch!

Comment (bebehei) [reply src/utils.c]: > I have no clue why this function needs to free only a but that's how it works it's best to preserve that until we can figure out why.

It's cool to have such a function to concatenate some stuff to something existing without having to check for anything:

```
n->msg = string_append(n->msg, addition, "\n");
```

In one line, you appended it if it's a non-empty and not-null string.

Comment (bebehei) [reply src/utils.c]: > a isn't freed in this case if it's \0 which could cause an (albeit minor) memory leak.

Now after freeing `a`, I found a missing `g_strdup` for the first parameter in a unittest. Thanks for that good catch.

**Comment by bebehei** [comment]
IMO this code should get merged now.

All invocations of `g_strconcat` are not this easily replaceable with `string_append`.

**Comment by tsipinakis** [comment]
LGTM

**Pull request merged**